roles/keystone/tasks/main.yml

```diff
@@ -31,9 +31,49 @@
     - etype: other
       permission: r
 
-- name: install keystone service
-  upstart_service: name=keystone user=keystone cmd=/usr/local/bin/keystone-all
-                   config_dirs=/etc/keystone
+- name: install keystone uwsgi service
+  template: src=roles/keystone/templates/etc/init/keystone.conf
```

Comment:
You don't need `roles/keystone/templates/` here. You can just reference `etc/init/keystone.conf`. Ansible looks in the role's templates/ directory for any paths first.